src/main/java/core/basesyntax/services/handlers/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.services.handlers.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.services.handlers.OperationHandler;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void getOperation(String fruit, int purchasedQuantity) {
+        for (Map.Entry<String, Integer> fruitTypeAndAmount : Storage.getFruitsTypeAndAmount()
```

Comment:
remove loops in handlers, don't iterate anything. Use (create if you don't have them) storage/DAO methods for that